### PR TITLE
Part --> CompoundFilter

### PR DESCRIPTION
**Pull request by berndhahnebach**
see forum topic:
http://forum.freecadweb.org/viewtopic.php?f=8&t=16773

especially starting from post:
http://forum.freecadweb.org/viewtopic.php?f=8&t=16773&start=10#p159031

cheers bernd

I have no glue what to do in the regard of all these new cool icon stuff. I just added one icon.

### CODE REVIEW

File: src/Mod/Part/CompoundTools/CompoundFilter.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "CompoundTools.CompoundFilter"
+__author__ = "DeepSOIC, Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+__doc__ = "Compound Filter: remove some childs from a compound (features)."
```
Comment (luzpaz) [inline src/Mod/Part/CompoundTools/CompoundFilter.py]: s/childs/children/

File: src/Mod/Part/CompoundTools/__init__.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "CompoundTools package"
+__author__ = "DeepSOIC, Bernd Hahnebach"
+__url__ = "http://www.freecadweb.org"
+__doc__ = """CompoundTools Package (part of FreeCAD)."""
+
+## @package CompoundTools
+#  \ingroup PART
```
Comment (luzpaz) [inline src/Mod/Part/CompoundTools/__init__.py]: `#  \brief CompoundTools Package for Part workbench`

Comment (berndhahnebach) [inline src/Mod/Part/CompoundTools/__init__.py]: done see rebase under this comment ...

**Comment by wwmayer** [comment]
I consider this problematic: `Base::Interpreter().runString("__import__('CompoundTools').importAll()");` because:
* it won't work with Python 3 because there the import mechanism works differently. CompoundTools are imported in Part but at this point the Part module is not yet initialized. Now CompoundTools tries to load Part too and then the initPart is called again which again tries to load CompoundTools and so on. This is a cyclic dependency and leads to a stack overflow
* CompoundTools is part of Part (not PartGui) but also deals with GUI stuff. In core modules I want to have a strict separation between App (i.e. non-GUI) and Gui stuff.

**Comment by berndhahnebach** [comment]
Ahh I got your point but I do not yet have an idea how to do this better. This part was copied from uptodate master. The BOPTools use the same code already. See https://github.com/FreeCAD/FreeCAD/blob/1463a26b70791f017327f5f8ffa9c6aef9850ce1/src/Mod/Part/App/AppPart.cpp#L370

bernd

**Comment by wwmayer** [comment]
> The BOPTools use the same code already. See

Exactly! That's why I know it leads to problems there. IMO, there is no need to put this import statement into the init function. It can e.g. also be put into the Initialize() method of the Workbench class (see InitGui.py).

**Comment by berndhahnebach** [comment]
https://github.com/FreeCAD/FreeCAD/pull/531

**Comment by wwmayer** [comment]
Replaced by #531

**Pull request closed without merging**